up-transport-vsomeip/src/lib.rs

```diff
@@ -166,81 +183,584 @@ impl UPTransportVsomeip {
     ///                             Should be set to `IP:port` of the endpoint mDevice
     /// * `ue_id` - the ue_id of the uEntity
     pub fn new(
-        authority_name: &AuthorityName,
+        uri: UUri,
         remote_authority_name: &AuthorityName,
-        ue_id: UeId,
         runtime_config: Option<RuntimeConfig>,
     ) -> Result<Self, UStatus> {
-        Self::new_internal(
-            authority_name,
-            remote_authority_name,
-            ue_id,
-            None,
-            runtime_config,
-        )
+        Self::new_internal(uri, remote_authority_name, None, runtime_config)
     }
 
     /// Creates a UPTransportVsomeip whether a vsomeip config file was provided or not
     fn new_internal(
-        authority_name: &AuthorityName,
+        uri: UUri,
         remote_authority_name: &AuthorityName,
-        ue_id: UeId,
         config_path: Option<&Path>,
         runtime_config: Option<RuntimeConfig>,
     ) -> Result<Self, UStatus> {
-        let optional_config_path: Option<PathBuf> = config_path.map(|p| p.to_path_buf());
+        uri.verify_rpc_response().map_err(|e| {
+            UStatus::fail_with_code(
+                UCode::INVALID_ARGUMENT,
+                format!("uri provided to transport is incorrect: {e:?}"),
+            )
+        })?;
 
         let (runtime_handle, thread_handle, shutdown_runtime_tx) =
             get_callback_runtime_handle(runtime_config);
 
-        let transport_inner: Arc<UPTransportVsomeipInnerHandle> = Arc::new({
-            if let Some(config_path) = optional_config_path {
-                let config_path = config_path.as_path();
-                let transport_inner_res = UPTransportVsomeipInnerHandle::new_with_config(
-                    authority_name,
-                    remote_authority_name,
-                    ue_id,
-                    config_path,
-                    runtime_handle.clone(),
+        let storage = Arc::new(UPTransportVsomeipStorage::new(
+            uri.clone(),
+            remote_authority_name.clone(),
+            runtime_handle.clone(),
+        ));
+
+        let engine = UPTransportVsomeipEngine::new(uri, None);
+        let point_to_point_listener = RwLock::new(None);
+        let optional_config_path: Option<PathBuf> = config_path.map(|p| p.to_path_buf());
+
+        Ok(Self {
+            storage,
+            engine,
+            point_to_point_listener,
+            config_path: optional_config_path,
+            thread_handle: Some(thread_handle),
+            shutdown_runtime_tx,
+        })
+    }
+    async fn await_engine(
```

Comment:
maybe add a newline in between?

Reply:
`cargo fmt` actually removes it if I try to add it :rofl:

Reply:
Not done, due to `cargo fmt` :rofl: